### PR TITLE
Feature/problems-microservice

### PR DESCRIPTION
**Pull request by aadua20**

### CODE REVIEW

**Comment by zkenshinx** [comment]
@sskhi1 Time to find bugs..

File: problems/src/main/java/com/freeuni/macs/model/ProblemId.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.freeuni.macs.model;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Data;
+
+@Schema(description = "Problem identifier")
+@Data
+public class ProblemId {
```
Comment (sskhi1) [inline problems/src/main/java/com/freeuni/macs/model/ProblemId.java:8]: This name confuses me 😣😣😣

File: problems/src/main/java/com/freeuni/macs/exception/ProblemExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice("com.freeuni.macs")
+@Slf4j
+public class ProblemExceptionHandler {
```
Comment (sskhi1) [inline problems/src/main/java/com/freeuni/macs/exception/ProblemExceptionHandler.java:14]: Great!

**Review by sskhi1 — APPROVED** [review]

File: problems/src/main/java/com/freeuni/macs/model/Problem.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private String name;
+
+    @Field(name = "description")
+    private String description;
+
+    @Field(name = "difficulty")
+    private String difficulty;
+
+    @Field(name = "topics")
+    private List<String> topics; // TODO: Create Topic class
```
Comment (sskhi1) [inline problems/src/main/java/com/freeuni/macs/model/Problem.java:33]: hmmmmmmmmmmmmmmmmmmmmmmmmm

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -9,4 +9,10 @@ services:
     build:
       context: ../macscode
     ports:
-      - "8080:8080"
\ No newline at end of file
+      - "8080:8080"
+  problems-db:
```
Comment (zkenshinx) [inline docker/docker-compose.yml:13]: Let's add login and password

File: problems/src/main/java/com/freeuni/macs/controller/ProblemController.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+@Tag(name = "Problems", description = "The Problem API")
+@RestController()
+@RequestMapping("/problems")
+public class ProblemController {
+    private final ProblemService problemService;
+
+    @Autowired
+    public ProblemController(ProblemService problemService) {
+        problemService.generateAndInsertRandomProblems(10);
```
Comment (zkenshinx) [inline problems/src/main/java/com/freeuni/macs/controller/ProblemController.java]: If we want to insert some initial data to database it's a better practice to use `CommandLineRunner` in config class:
```java
@Bean
public CommandLineRunner initializeData(ProblemService problemService) {
     return args -> {
           problemService.generateAndInsertRandomProblems(10);
     };
}
```

File: problems/src/main/java/com/freeuni/macs/model/ProblemId.java (hunk 1 above)
Comment (aadua20) [reply problems/src/main/java/com/freeuni/macs/model/ProblemId.java:8]: me t👀

File: problems/pom.xml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+			<optional>true</optional>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+
```
Comment (zkenshinx) [inline problems/pom.xml]: Why do we have 2 spaces?

File: docker/docker-compose.yml (hunk 4 above)
Comment (sskhi1) [inline docker/docker-compose.yml:13]: add container name please.

File: problems/src/test/java/com/freeuni/macs/ProblemsApplicationTests.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class ProblemsApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
```
Comment (zkenshinx) [inline problems/src/test/java/com/freeuni/macs/ProblemsApplicationTests.java:11]: These are the best tests ever

**Review by zkenshinx — CHANGES_REQUESTED** [review]

**Review by zkenshinx — APPROVED** [review]

**Pull request merged**